### PR TITLE
Add possibility to translate strings

### PR DESCRIPTION
**Pull request by jaysonsantos**
To integrate with flask_babel it is simple as doing this:

```python
from flask_babel import lazy_gettext
from schematics.translator import register_translator
register_translator(lazy_gettext)
```

### CODE REVIEW

**Comment by lkraider** [comment]
Thanks for the patch, we've been missing i18n support for a while (had a patch on 0.9 but never integrated). I'll review this and merge when approved.

**Comment by jaysonsantos** [comment]
The way I did is not enforce any tool. We use flask_babel here and to extract the strings we just use `pybabel extract schematics_folder` and translated by ourselves. I *think* Transifex has a free option for open source projects where community could help translating the project.

File: schematics/translator.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class LazyText(object):
```
Comment (lkraider) [inline schematics/translator.py]: Please add these lines to make the class unicode-compat aware:
```python
# -*- coding: utf-8 -*-

from .compat import *


@str_compat
class LazyText(object):
```

File: schematics/types/base.py [hunk 2]
```diff
@@ -321,7 +322,7 @@ def check_required(self, value, context):
     def validate_choices(self, value, context):
         if self.choices is not None:
             if value not in self.choices:
-                raise ValidationError(self.messages['choices']
+                raise ValidationError(str(self.messages['choices'])
```
Comment (jaysonsantos) [inline schematics/types/base.py]: good idea, I will implement it

File: schematics/types/compound.py [hunk 3]
```diff
@@ -128,7 +129,7 @@ def _convert(self, value, context):
             model_class = self.model_class
         else:
             raise ConversionError(
-                "Input must be a mapping or '%s' instance" % self.model_class.__name__)
+                _("Input must be a mapping or '%s' instance", lazy=False) % self.model_class.__name__)
```
Comment (jaysonsantos) [inline schematics/types/compound.py]: if we implement `__mod__` and `format` and we use it we should be ok

File: schematics/types/base.py [hunk 4]
```diff
@@ -394,24 +393,24 @@ def to_native(self, value, context=None):
...
             else:
                 return str(value)
-        raise ConversionError(str(self.messages['convert']).format(value))
+        raise ConversionError(self.messages['convert'].format(value))
 
     def validate_length(self, value, context=None):
         length = len(value)
         if self.max_length is not None and length > self.max_length:
-            raise ValidationError(self.messages['max_length'])
+            raise ValidationError(str(self.messages['max_length']))
```
Comment (lkraider) [inline schematics/types/base.py]: Are these casts required, since you are casting inside exceptions.FieldError ?

File: schematics/translator.py [hunk 5]
```diff
@@ -1,3 +1,6 @@
+# coding: utf-8
+
+
 class LazyText(object):
```
Comment (lkraider) [inline schematics/translator.py]: The import and class decorator are still missing.

File: schematics/types/base.py (hunk 4 above)
Comment (jaysonsantos) [inline schematics/types/base.py]: I missed some, I will fix it now.

**Comment by lkraider** [comment]
Perfect, thanks so much for the contribution!

**Pull request merged**